# "-march=native" does not work for RISCV so we need to set this flag properly
# author: julianmorillo
--- metaeuk-6-a5d39d9/lib/mmseqs/CMakeLists.txt.orig	2025-03-07 17:53:34.646002874 +0100
+++ metaeuk-6-a5d39d9/lib/mmseqs/CMakeLists.txt	2025-03-07 17:57:44.471819266 +0100
@@ -90,6 +90,8 @@
         set(ZARCH 1)
     elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "^sparc")
         set(SPARC 1)
+    elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "riscv64")
+        set(RISCV64 1) 
     else ()
         message(WARNING "CPU without native SIMD instructions. Performance will be bad.")
     endif ()
@@ -113,6 +115,8 @@
         set(MMSEQS_ARCH "-mcpu=native")
     elseif (ZARCH)
         set(MMSEQS_ARCH "-mcpu=native -mzvector")
+    elseif (RISCV64)
+	 execute_process(
+               COMMAND bash "-c" "cat /proc/cpuinfo | grep isa | head -1 | cut -d':' -f2"
+               OUTPUT_VARIABLE RISCV64_ISA_RAW
+               OUTPUT_STRIP_TRAILING_WHITESPACE
+        )
+        string(STRIP ${RISCV64_ISA_RAW} RISCV64_ISA)
+        set(MMSEQS_ARCH "-march=${RISCV64_ISA}")
     else ()
         set(MMSEQS_ARCH "-march=native")
     endif ()
